src/Text/Parsing/StringParser/String.purs

```diff
@@ -111,3 +115,39 @@ anyLetter = lowerCaseChar <|> upperCaseChar <?> "Expected a letter"
 -- | Match a letter or a number.
 alphaNum :: Parser Char
 alphaNum = anyLetter <|> anyDigit <?> "Expected a letter or a number"
+
+-- | Build the regular expression from the pattern and match it, ensuring
+-- | that the pattern only attempts to match from the start of the target.
+regex' :: String -> Parser String
+regex' pat =
+    case Regex.regex pattern noFlags of
+      Left _ ->
+        fail $ "Text.Parsing.StringParser.String.regex': illegal regex " <> pat
+      Right r ->
+        regex r
+    where
+      pattern =
+        case stripPrefix (Pattern "^") pat of
+          Nothing ->
+            "^" <> pat
+          _ ->
+            pat
+
+-- | Match the regular expression.
+regex :: Regex.Regex -> Parser String
+regex r =
+  Parser \{ str, pos } ->
+    let
+      remainder = drop pos str
+    in
+      -- reduce the possible array of matches to 0 or 1 elements to aid Array pattern matching
+      case uncons $ fromMaybe [] $ Regex.match r remainder of
+        Just { head: Just matched, tail: _ }  ->
+          -- only accept matches at position 0
+          case stripPrefix (Pattern matched) remainder of
```

Comment:
OK - I think I see what you're saying.  A future version of the String module could memoise the regex - is that right?  (I had been assuming it was the responsibility of the client).  Anyway, are you OK for me to swap the names of regex and regex', only expose what is now regex and just leave it at that?